src/components/Dialogs/ImageCropDialog/cropper.ts

```diff
@@ -44,10 +46,13 @@ const CANVAS_OPTS_PER_TYPE: Record<CropperImageType, Cropper.GetCroppedCanvasOpt
   },
 }
 
-export const useCropper = ({ imageEl, imageType }: UseCropperOpts) => {
+export type CropBoxData = Cropper.CropBoxData
+
+export const useCropper = ({ imageEl, imageType, fileName }: UseCropperOpts) => {
   const [cropper, setCropper] = useState<Cropper | null>(null)
   const [currentZoom, setCurrentZoom] = useState(0)
   const [zoomRange, setZoomRange] = useState<[number, number]>([0, 1])
+  const { uploadingFilesData } = useUploadingFilesData()
```

Comment:
We cannot really rely on this here. A couple of reasons:
- We create a weird link - `ImageCropDialog` now relies on `useUploadingFilesData` and will actually throw an error if used outside of `UploadingFilesDataProvider`. This doesn't make much sense since one doesn't have anything to do with the other. We need to separate those concerns 
- Consider what's happening in the `useEffect` now. It has `uploadingFilesData` as a dependency. So whenever the data about files being uploaded changes, the hook will re-run, creating a new cropper instance. For example, let's say I started uploading a video. It's processing in the background and now I want to upload another one. So I pick a thumbnail and start cropping it. If during my cropping, one of the files from the previous videos finishes uploading, my cropping flow will be interrupted and reset. That's not a great user experience

Actually, maybe let's try figuring this restoration flow later on. It will not be that simple, the image for cropping would need to be displayed somewhere, we will probably need a custom hook for that